src/async_impl/body.rs

```diff
@@ -0,0 +1,98 @@
+use futures::{Stream, Poll, Async};
+use bytes::Bytes;
+
+pub struct Body {
```

Comment:
Indeed, but asynchronous bodies need to be `Stream`, and that means `File` and `Box<Read>` and friends don't fit, yet. I wanted to get the upgrade done before figuring out all that.